### DISCUSSION TITLE
Why is Cypress the recommended E2E framework?

### DISCUSSION THREAD

**Discussion by wederer** [vuejs/docs#2309] [General]
In the docs under [E2E Testing](https://vuejs.org/guide/scaling-up/testing.html#recommendation-2) Cypress is recommended as the preferred E2E testing framework while pretty much any article/blog that compares Cypress to Playwright seems to prefer Playwright for a multitute of reasons (e.g. [here](https://mtlynch.io/notes/cypress-vs-playwright/) or [here](https://applitools.com/blog/cypress-vs-playwright/)).  After testing both frameworks I would also agree with these results and prefer Playwright.

Has the decision by the Vue.js team been documented somewhere? Maybe a blog article?

**Comment by splincool** [comment]
Maybe because at the time of writing that section of documentation Cypress was more popular than Playwright. Also, Cypress provides support for Component Testing and Playwright only has it as experimental at this time.

**Comment by Shyam-Chen** [comment]
Playwright has recently introduced Component Testing, but it is still in an experimental stage.
